src/RazorSdk/SourceGenerators/RazorSourceGeneratorEventSource.cs

```diff
@@ -0,0 +1,137 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics.Tracing;
+
+namespace Microsoft.NET.Sdk.Razor.SourceGenerators
+{
+    [EventSource(Name = "Microsoft.NET.Sdk.Razor.SourceGenerators.RazorSourceGenerator")]
+    internal sealed class RazorSourceGeneratorEventSource : EventSource
+    {
+        public static readonly RazorSourceGeneratorEventSource Log = new();
+
+        private RazorSourceGeneratorEventSource() { }
+
+        private const int ComputeRazorSourceGeneratorOptionsId = 1;
+
+        [Event(ComputeRazorSourceGeneratorOptionsId, Level = EventLevel.Informational)]
+        public void ComputeRazorSourceGeneratorOptions()
+        {
+            if (IsEnabled())
```

Comment:
Each `[Event]` method is going to call `IsEnabled` as soon as it enters `WriteEvent`, so you don't need to bother wrapping the call to `WriteEvent` in the if check.  If the computation of the arguments is expensive, then you should wrap that computation and the call to write the event in a call to `IsEnabled`.